### PR TITLE
Add the concept of streams to Code Intelligence

### PR DESCRIPTION
**Pull request by scalabilitysolved**
Code Intelligence has been working for some time with the concept of streams (multiple distinct working groups within the main team), this is the start of codifying this in a lightweight way.

I also took the opportunity to trim down the Code Intelligence page to focus only on how we're working now and moved the history section into its own page.  

Review live: https://deploy-preview-4569--sourcegraph-handbook.netlify.app/departments/engineering/teams/code-intelligence/

### CODE REVIEW

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 1]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
+- People feel 100% supported to comment, help and collaborate across the whole team.
 
-## How we do technical hiring
+There is no "that's not my responsibility" with the use of streams, we just have certain areas that are best filled by specific skill sets.
 
-- [Code Walkthrough](code-walkthrough.md)
+### Language Tools
 
-## Team communication
+- Slack channel: #language-tools
```
Comment (varungandhi-src) [inline content/departments/engineering/teams/code-intelligence/index.md]:
```suggestion
- Slack channel: #language-tools-internal
```

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 2]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
 
-**Architecture Presentation**: An experienced code intel engineer will give a live presentation primarily to the new hire (and any other Sourcegraph teammate who’s interested) explaining the architecture of our systems at a high level.
+### Code Navigation
 
-**Pair Programming**: Keeping inline with [engineering pairing sessions](../../dev/onboarding/index.md#pairing-sessions) we will support our new teammates by setting aside dedicated time every week for mentoring. This not only helps deliver guided learning but also is great for team and relationship building.
+Code Navigation owns the UI/UX customer facing code navigation experience predominately focusing on the file view.
 
-**First Tasks**: The team has marked issues we believe are good first issues in [Github](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+label%3A%22good+first+issue%22++label%3A%22team%2Fcode-intelligence%22), following the [starter task process](../../dev/onboarding/index.md#starter-tasks) work with the code intel engineering manager to select three issues that you think are good ones for you to work on in your first 4 weeks.
+- Slack channel: #code-navigation
```
Comment (varungandhi-src) [inline content/departments/engineering/teams/code-intelligence/index.md]:
```suggestion
- Slack channel: #code-navigation-internal
```

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 3]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
+### Language Platform
 
-  1. Add any agenda items that should be discussed.
-  1. Review any existing agenda items and be prepared to discuss them.
+Language Platform's focus is split into two distinct areas, the creation of data (invoking SCIP indexers via Executors _or_ generating symbols) and the productionalization of said data (exposing the data in a queryable, performant and scalable way). Language Platform aims to empower two users, providing a programatic/pluggable system for the Language Tools stream and for downstream clients to build ontop of the Code Graph data (the Code Navigation stream and also additional product teams).
 
-- Team Updates: Each day, our Daily Standup Bot prompts us to add a text check-in by answering what we worked on the day before, what we're planning to work on next, and if there are any current blockers.
+- Slack channel: #language-platform
```
Comment (varungandhi-src) [inline content/departments/engineering/teams/code-intelligence/index.md]:
```suggestion
- Slack channel: #language-platform-and-navigation-internal
```

File: content/departments/engineering/teams/code-intelligence/history.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+## History
+
+The origin of our work is the core of Sourcegraph, first implemented as [srclib](https://github.com/sourcegraph/srclib) a code analysis tool developed by our co-founders [Beyang](https://github.com/beyang) and [Quinn](https://github.com/sqs) in 2015.
+
+2016 - We evolved our strategy to utilize LSP (Language Server Protocol) to power the new Sourcegraph code search engine. LSP is a user oriented protocol and requires an implementation of a language server for each programming language.
+
+2017 - Over time we started to experience some issues with language servers. They can be difficult to deploy, slow at runtime, slow to adopt by members of their respective language communities, and slow to develop.
+
+2018 - We added a new basic code intelligence that is built on search-based heuristics to allow us to provide quick and good enough i.e. "imprecise" support for the [most popular programming languages](https://sourcegraph.com/extensions?query=category%3A%22Programming+languages%22).
```
Comment (macraig) [inline content/departments/engineering/teams/code-intelligence/history.md]: The link to extensions will soon be deprecated, we'll need to link to a new doc with the list of supported languages

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 5]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
 
-- [Code Walkthrough](code-walkthrough.md)
+### Language Tools
 
-## Team communication
+- Slack channel: #language-tools-internal
+- Github board --> https://github.com/orgs/sourcegraph/projects/211/views/51
 
-Here are some key ways to contact us:
+##### Current members
```
Comment (macraig) [inline content/departments/engineering/teams/code-intelligence/index.md:55]: Should we link to their handbook page or include full names?

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 6]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
-- On Slack, use the #code-intel channel or tag the @codeintel team.
+- Olaf
+- TJ
+- Varun
 
-## Team process
+##### Focus and ownership
 
-The code intel team has defined a set of rituals that keeps the team connected around our common goals. As a team we also believe in good meeting etiquette, and to ensure effective meetings for everyone every meeting must have a meaningful topic, agenda, and stated goal or purpose statement defined ahead of time.
+The Language Tools stream focus on tooling specific to programming languages and their counterpart ecosystems to provide the best Precise Intelligence experience. The aim is to have a symbiotic relationship
```
Comment (macraig) [inline content/departments/engineering/teams/code-intelligence/index.md]:
```suggestion
The Language Tools stream focus on tooling specific to programming languages and their counterpart ecosystems to provide the best precise code navigation experience. The aim is to have a symbiotic relationship
```

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 7]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
+with the Language Platform team in that, Language Tools will provide the specific logic per problem (think syntax highlighting) whilst Language Platform will expose an agnostic/programatic way to scale this. Specific areas of ownership include:
 
-Our iterations last two weeks and are comprised of the following:
+- Precise Intelligence
+  - [SCIP](https://github.com/sourcegraph/scip#readme)
+  - SCIP indexers
```
Comment (macraig) [inline content/departments/engineering/teams/code-intelligence/index.md]:
```suggestion
  - [SCIP indexers](https://docs.sourcegraph.com/code_intelligence/references/indexers)
```

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 8]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
-- Team Sync Meeting: The second Monday of the iteration, the code intelligence team meets to stay in sync and hear from stakeholders. We use a Google doc for [agenda and meeting notes](https://docs.google.com/document/d/1R4gXavKwajVRplHSy1ECn_ZHMoQZIwiGKqWWb2SdbUE/edit). Prior to the sync meeting, each team member should:
+### Language Platform
 
-  1. Add any agenda items that should be discussed.
-  1. Review any existing agenda items and be prepared to discuss them.
+Language Platform's focus is split into two distinct areas, the creation of data (invoking SCIP indexers via Executors _or_ generating symbols) and the productionalization of said data (exposing the data in a queryable, performant and scalable way). Language Platform aims to empower two users, providing a programatic/pluggable system for the Language Tools stream and for downstream clients to build ontop of the Code Graph data (the Code Navigation stream and also additional product teams).
```
Comment (macraig) [inline content/departments/engineering/teams/code-intelligence/index.md]:
```suggestion
Language Platform's focus is split into two distinct areas, the creation of data (invoking SCIP indexers via Executors _or_ generating symbols) and the productionalization of said data (exposing the data in a queryable, performant and scalable way). Language Platform aims to empower two users, providing a programatic/pluggable system for the Language Tools stream and for downstream clients to build on top of the Code Graph data (the Code Navigation stream and also additional product teams).
```

**Review by macraig — APPROVED** [review]

File: content/departments/engineering/teams/code-intelligence/history.md (hunk 4 above)
Comment (scalabilitysolved) [reply content/departments/engineering/teams/code-intelligence/history.md]: Widening the question further, is this page even useful?  I just pulled it out of the previous page we had, we could just delete the whole page etc.  If not I can change to link to some other docs page.

File: content/departments/engineering/teams/code-intelligence/index.md (hunk 5 above)
Comment (scalabilitysolved) [reply content/departments/engineering/teams/code-intelligence/index.md:55]: I can change to full names, I'll see if I can link to the handbook page for people but that data is generated from yaml files dynamically.

File: content/departments/engineering/teams/code-intelligence/history.md (hunk 4 above)
Comment (macraig) [reply content/departments/engineering/teams/code-intelligence/history.md]: Do you mean if having a list of search-based supported languages is useful? My understanding is that CEs find it useful to consult (see https://sourcegraph.slack.com/archives/C022SPMNR0W/p1659640668630779)

Comment (scalabilitysolved) [reply content/departments/engineering/teams/code-intelligence/history.md]: I mean this history page.

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 9]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
 
-## How we do technical hiring
+There is no "that's not my responsibility" with the use of streams, we just have certain areas that are best filled by specific skill sets.
 
-- [Code Walkthrough](code-walkthrough.md)
+### Language Tools
 
-## Team communication
+- Slack channel: #language-tools-internal
+- Github board --> https://github.com/orgs/sourcegraph/projects/211/views/51
```
Comment (varungandhi-src) [inline content/departments/engineering/teams/code-intelligence/index.md]: (Apologies for pedantry)

```suggestion
- GitHub board: https://github.com/orgs/sourcegraph/projects/211/views/51
```

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 10]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
+### Language Platform
 
-  1. Add any agenda items that should be discussed.
-  1. Review any existing agenda items and be prepared to discuss them.
+Language Platform's focus is split into two distinct areas, the creation of data (invoking SCIP indexers via Executors _or_ generating symbols) and the productionalization of said data (exposing the data in a queryable, performant and scalable way). Language Platform aims to empower two users, providing a programatic/pluggable system for the Language Tools stream and for downstream clients to build on top of the Code Graph data (the Code Navigation stream and also additional product teams).
 
-- Team Updates: Each day, our Daily Standup Bot prompts us to add a text check-in by answering what we worked on the day before, what we're planning to work on next, and if there are any current blockers.
+- Slack channel: #language-platform-and-navigation-internal
+- Github board --> https://github.com/orgs/sourcegraph/projects/211/views/48
```
Comment (varungandhi-src) [inline content/departments/engineering/teams/code-intelligence/index.md:80]:
```suggestion
- GitHub board: https://github.com/orgs/sourcegraph/projects/211/views/48
```

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 11]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
+### Code Navigation
 
-**Pair Programming**: Keeping inline with [engineering pairing sessions](../../dev/onboarding/index.md#pairing-sessions) we will support our new teammates by setting aside dedicated time every week for mentoring. This not only helps deliver guided learning but also is great for team and relationship building.
+Code Navigation owns the UI/UX customer facing code navigation experience predominately focusing on the file view.
 
-**First Tasks**: The team has marked issues we believe are good first issues in [Github](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+label%3A%22good+first+issue%22++label%3A%22team%2Fcode-intelligence%22), following the [starter task process](../../dev/onboarding/index.md#starter-tasks) work with the code intel engineering manager to select three issues that you think are good ones for you to work on in your first 4 weeks.
+- Slack channel: #code-navigation-internal
+- Github board --> https://github.com/orgs/sourcegraph/projects/211/views/21
```
Comment (varungandhi-src) [inline content/departments/engineering/teams/code-intelligence/index.md]:
```suggestion
- GitHub board: https://github.com/orgs/sourcegraph/projects/211/views/21
```

**Review by varungandhi-src — APPROVED** [review]

File: content/departments/engineering/teams/code-intelligence/history.md (hunk 4 above)
Comment (macraig) [reply content/departments/engineering/teams/code-intelligence/history.md]: Personally I would delete it

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 12]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
 
-Our iterations last two weeks and are comprised of the following:
+- Precise Intelligence
+  - [SCIP](https://github.com/sourcegraph/scip#readme)
+  - [SCIP indexers](https://docs.sourcegraph.com/code_intelligence/references/indexers)
+  - LSIF indexers
+  - Community indexer contributions
+- Auto-inference behavior for indexing projects
+- Syntax highlighting (including the syntect service)
+- Package integrations (Maven,NPM, PyPI, Crates)
```
Comment (efritz) [inline content/departments/engineering/teams/code-intelligence/index.md]:
```suggestion
- Package integrations (Maven, NPM, PyPI, Crates)
```

**Review by efritz — APPROVED** [review]

File: content/departments/engineering/teams/code-intelligence/index.md (hunk 12 above)
Comment (varungandhi-src) [reply content/departments/engineering/teams/code-intelligence/index.md]: (Apologies for pedantry again but:) It should be `npm` not NPM.

**Review by Strum355 — APPROVED** [review]

File: content/departments/engineering/teams/code-intelligence/index.md [hunk 13]
```diff
@@ -10,115 +12,161 @@ The Code Intelligence team builds tools and services that provide contextual inf
...
+## Members
 
-## Direction and strategy
+{{generator:product_team.code_intelligence}}
 
-See [Code intelligence strategy](../../../../strategy-goals/strategy/code-intelligence/index.md)
+## How we work
 
-## Overview
+On Slack, use the `#code-intel` channel, tag `@code-intel-support` for any help or use the `@codeintel` handle to notify the whole team. As detailed below in the streams section we do have several additional internal rooms however we expect almost all questions and conversations with external teams/departments to take place in `#code-intel`.
```
Comment (cesrjimenez) [inline content/departments/engineering/teams/code-intelligence/index.md:33]: Are we still #code-intel even after the new marketing efforts?

**Review by cesrjimenez — APPROVED** [review]

File: content/departments/engineering/teams/code-intelligence/history.md (hunk 4 above)
Comment (scalabilitysolved) [reply content/departments/engineering/teams/code-intelligence/history.md]: WIll remove 👍

File: content/departments/engineering/teams/code-intelligence/index.md (hunk 13 above)
Comment (scalabilitysolved) [reply content/departments/engineering/teams/code-intelligence/index.md:33]: Correct, internally we still are.

File: content/departments/engineering/teams/code-intelligence/index.md (hunk 12 above)
Comment (scalabilitysolved) [reply content/departments/engineering/teams/code-intelligence/index.md]: No need to apologise, will change to `npm` :)

**Review by tjdevries — COMMENTED** [review]
LGTM (read through but didn't find any comments before)

**Pull request merged**